File: apps/web/features/analyze/agents.ts
export interface Agent {
  id: string;
  icon: string;
  name: string;
  role: string;
  mock: string;
}

export const AGENTS: Agent[] = [
  {
    id: "cfo",
    icon: "💰",
    name: "CFO",
    role: "Financial",
    mock: "Financial projections show potential $50M market cap by year 3. Target burn rate of $200K/month for the first 18 months, with a death valley crossing at month 14. Recommend a Series A raise of $3M at a $15M pre-money valuation after hitting 1,000 paying customers. Unit economics need to reach CAC payback under 12 months.",
  },
  {
    id: "cmo",
    icon: "📣",
    name: "CMO",
    role: "Marketing",
    mock: "Go-to-market strategy should focus on product-led growth targeting SMBs first. Organic content + community can drive early traction. Target CAC below $150 with an LTV:CAC ratio of 4:1. Channel mix: 40% organic, 35% paid search, 25% partnerships. Launch on ProductHunt + targeted LinkedIn outreach to 500 ICP accounts.",
  },
  {
    id: "market",
    icon: "📊",
    name: "Market Analyst",
    role: "Research",
    mock: "TAM estimated at $12B globally with a SAM of $800M for the English-speaking B2B segment. Key competitors are fragmented, leaving a clear opening for a well-executed vertical play. Market growing at 24% CAGR. Top 3 competitors have NPS below 30 — customer satisfaction gap is the main wedge.",
  },
  {
    id: "vc",
    icon: "🏦",
    name: "VC Investor",
    role: "Investment",
    mock: "Strong founder-market fit potential. Seed milestones: working MVP, 10 design partners, $10K MRR. Series A trigger: $100K MRR with net revenue retention above 110%. Exit scenarios include strategic acquisition at 5–8x revenue or IPO at $300M+ ARR. Comparable exits in this space averaged 7.2x revenue.",
  },
  {
    id: "devil",
    icon: "⚠️",
    name: "Devil's Advocate",
    role: "Risk",
    mock: "Critical risks: regulatory headwinds in the EU could delay expansion 12–18 months. CAC may spike as incumbents respond aggressively. Technology moat is thin — a well-funded competitor could replicate core features in 6 months. Default scenario: if growth stalls below plan at month 18, runway collapses without bridge round.",
  },
  {
    id: "pm",
    icon: "🚀",
    name: "Product Manager",
    role: "Roadmap",
    mock: "MVP scope: core value prop in 3 user flows, shipped in 8 weeks. Phase 1 KPIs: 10-minute time-to-value, 60% D7 retention. Prioritize Slack and Google Workspace integrations for distribution leverage. V1.1 adds team collaboration features based on design partner feedback. Technical debt budget: max 20% of each sprint.",
  },
];

File: apps/api/agents.py
from dataclasses import dataclass

@dataclass(frozen=True)
class AgentDef:
    id: str
    name: str
    role: str
    system_prompt: str


AGENTS: list[AgentDef] = [
    AgentDef(
        id="cfo",
        name="CFO",
        role="Financial",
        system_prompt=(
            "You are a CFO and financial strategist. Analyze the given business idea "
            "from a financial perspective. Cover: funding requirements, burn rate "
            "projections, unit economics, revenue model, key financial risks, and path "
            "to profitability. Be concise and specific — 4–6 sentences. Use concrete "
            "numbers where reasonable."
        ),
    ),
    AgentDef(
        id="cmo",
        name="CMO",
        role="Marketing",
        system_prompt=(
            "You are a CMO and growth strategist. Analyze the given business idea from "
            "a marketing and go-to-market perspective. Cover: target customer profile, "
            "acquisition channels, positioning, CAC/LTV expectations, and launch "
            "strategy. Be concise and specific — 4–6 sentences."
        ),
    ),
    AgentDef(
        id="market",
        name="Market Analyst",
        role="Research",
        system_prompt=(
            "You are a Market Analyst. Analyze the given business idea from a market "
            "research perspective. Cover: TAM/SAM/SOM estimates, competitive landscape, "
            "market trends, customer pain points, and key differentiators. Be concise "
            "and specific — 4–6 sentences."
        ),
    ),
    AgentDef(
        id="vc",
        name="VC Investor",
        role="Investment",
        system_prompt=(
            "You are a VC Investor evaluating a potential investment. Analyze the given "
            "business idea from an investment perspective. Cover: founder-market fit "
            "indicators, key milestones for seed/Series A, exit potential, comparable "
            "companies, and top 2–3 risks. Be concise and specific — 4–6 sentences."
        ),
    ),
    AgentDef(
        id="devil",
        name="Devil's Advocate",
        role="Risk",
        system_prompt=(
            "You are a Devil's Advocate. Critically challenge the given business idea. "
            "Identify the 3–4 most serious risks, blind spots, or assumptions that "
            "could cause this to fail. Be direct and specific. Do not be encouraging — "
            "your job is to surface what could go wrong."
        ),
    ),
    AgentDef(
        id="pm",
        name="Product Manager",
        role="Roadmap",
        system_prompt=(
            "You are a Product Manager. Analyze the given business idea from a product "
            "strategy perspective. Cover: MVP scope, core user flows, success KPIs, key "
            "integrations, technical risks, and a realistic timeline. Be concise and "
            "specific — 4–6 sentences."
        ),
    ),
]

AGENT_MAP: dict[str, AgentDef] = {a.id: a for a in AGENTS}
